Health/Presentation/Personal/PersonalViewController.swift

```diff
@@ -25,22 +25,74 @@ class PersonalViewController: CoreGradientViewController, Alertable {
     private var networkService = DefaultNetworkService()
     private var didShowLocationPermissionAlert = false
 
+    private var distanceViewModel = CourseDistanceViewModel()
+    private var previousLocationPermission = false  // 이전 권한 상태 추적
 
     override func initVM() { }
 
     override func viewDidLoad() {
         super.viewDidLoad()
         setupDataSource()
         applyInitialSnapshot()
-        loadWalkingCourses()
-        Task {
+        setupDistanceViewModel()
+        previousLocationPermission = LocationPermissionService.shared.checkCurrentPermissionStatus()
+
+        NotificationCenter.default.addObserver(
+               self,
+               selector: #selector(checkLocationPermissionChange),
+               name: UIApplication.didBecomeActiveNotification,
+               object: nil
+           )
```

Comment:
SceneDelegate 에서는 제가 잘못된 위치에 선언했어요
```suggestion
        NotificationCenter.default.addObserver(
            self,
            selector: #selector(checkLocationPermissionChange),
            name: UIApplication.willEnterForegroundNotification,
            object: nil
        )
```